modules/lmdbbackend/lmdbbackend.cc

```diff
@@ -1297,56 +1306,62 @@ bool LMDBBackend::deleteDomain(const DNSName& domain)
     throw DBException(std::string(__PRETTY_FUNCTION__) + " called without a transaction");
   }
 
-  int transactionDomainId = d_transactiondomainid;
-  DNSName transactionDomain = d_transactiondomain;
-
   abortTransaction();
 
-  uint32_t id;
+  LMDBIDvec idvec;
 
-  { // get domain id
+  if (!d_handle_dups) {
+    // get domain id
     auto txn = d_tdomains->getROTransaction();
 
     DomainInfo di;
-    id = txn.get<0>(domain, di);
+    idvec.push_back(txn.get<0>(domain, di));
   }
+  else {
+    auto txn = d_tdomains->getROTransaction();
+
+    txn.get_multi<0>(domain, idvec);
```

Comment:
This part I do not follow. AFAIKS, it throws always

Reply:
get_multi pushes into idvec, does that help?

Reply:
I get that, but the throw below is unconditional...

Reply:
there was an if there, looks like i lost a commit during a rebase..